beacon-chain/state/stateV0/state_trie.go

```diff
@@ -31,7 +32,31 @@ var (
 
 // InitializeFromProto the beacon state from a protobuf representation.
 func InitializeFromProto(st *pbp2p.BeaconState) (*BeaconState, error) {
-	return InitializeFromProtoUnsafe(proto.Clone(st).(*pbp2p.BeaconState))
+	clonedState, err := CloneBeaconState(st)
+	if err != nil {
+		return nil, err
+	}
+	return InitializeFromProtoUnsafe(clonedState)
+}
+
+// CloneBeaconState serves as a wrapper for proto.Clone to apply any
+// post-cloning changes needed. This was added due to a change in protobuf v2
+// which encodes empty slices as nil.
+func CloneBeaconState(st *pbp2p.BeaconState) (*pbp2p.BeaconState, error) {
+	if st == nil {
+		return nil, errors.New("received nil state")
+	}
+	cloned, ok := proto.Clone(st).(*pbp2p.BeaconState)
+	if !ok {
+		return nil, errors.New("not a beacon state")
+	}
+	if cloned == nil {
+		return nil, errors.New("beacon state clone is empty")
+	}
+	if cloned.Validators == nil {
```

Comment:
why only the validators field. It could apply to many other fields

Reply:
Haven't encountered problems with others but will add more cases.

Reply:
Did this only come up in tests ?

Reply:
I think so, yes, let me confirm if it arises in runtime

Reply:
Confirmed, only in testing, got rid of these changes.